modules/renter/hostdb/hostdb.go

```diff
@@ -66,13 +66,11 @@ type HostDB struct {
 // a hostContract includes the original contract made with a host, along with
 // the most recent revision.
 type hostContract struct {
+	IP              modules.NetAddress
 	ID              types.FileContractID
 	FileContract    types.FileContract
 	LastRevisionTxn types.Transaction
 	SecretKey       crypto.SecretKey
-
-	// revisions must happen in serial
-	mu sync.RWMutex
```

Comment:
yeah. It's kind of a logic error for two threads to be revising the same contract anyway. So now the onus is on the renter not to call `Upload` concurrently on the same `Uploader` (it already avoids doing this anyway). Removing the mutex decreases complexity and prevents a lot of annoying pass-by-value errors ([as seen in the host](https://github.com/NebulousLabs/Sia/blob/master/modules/host/persist.go#L40)).